### DISCUSSION TITLE
Dynamical matrix in LAMMPS with DeepMD potential

### DISCUSSION THREAD

**Discussion by matr1x-1** [deepmodeling/deepmd-kit#3941] [Q&A]
Hello, 

I am trying to calculate a dynamical matrix with my potential in LAMMPS using the command: 

"dynamical_matrix all eskm 0.000001 file dynmat.dat binary no" 

However, this gives an error "ERROR: 6-element atomic virial is not supported. Use compute centroid/stress/atom command for 9-element atomic virial. (/home/conda/feedstock_root/build_artifacts/libdeepmd_1698439047902/work/source/lmp/pair_deepmd.cpp:470)". 

Strangely, when I do a minimization before it doesn't crash but I get negative eigenvalues, as described here: https://matsci.org/t/why-does-dynamical-matrix-return-negative-values/50729

The solution there was to enable double precision on the GPU. 

Has anyone else managed to successfully calculate the dynamical matrix with DeePMD? 

Thanks!

**Comment by njzjz** [comment]
We have fixed this bug in #3133. Please use a newer version.

**Reply by matr1x-1** [reply]
Thanks for the quick reply! Using a newer version fixed the problem with the 6-element atomic virial. The negative values still persist though. It could be that the model is just not good enough yet.
